docs/network-policy.md

```diff
@@ -0,0 +1,361 @@
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+If only ingress traffic to all pods in a namespace needs to be blocked the following can be used 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+```
+
+And finally if only egress traffic to all pods in a namespace needs to be blocked 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Egress
+```
+**OVN-Implementation:**
+
+Each NetworkPolicy creates a port group named `FOO_bar` where `FOO` is the policy's Namespace and `bar` is the policy's name.  All pods that the policy's `podSelector` selects are added to the port group.
+
+Two initial global deny PortGroups are used, specifially: `ingressDefaultDeny` and `egressDefaultDeny`.  Any pod selected by a NetworkPolicy in any namespace is added to these PortGroups.
+
+```
+    _uuid               : 1deeac49-e87e-4e05-9324-beb8ef0dcef4
+    acls                : [3f864884-cdb7-44be-a60e-e4f743afc9d0, f174fcf1-a7c2-496d-9b96-136aaccc014f]
+    external_ids        : {name=ingressDefaultDeny}
+    name                : ingressDefaultDeny
+    ports               : [ce1bc4e5-0309-463f-9fd1-80f6e487e2d4]
+
+    _uuid               : 5249b7a2-36bf-4246-98ea-13d5c5e17c68
+    acls                : [36ed4154-71ab-4b8f-8119-5c4cc92708d9, c6663e29-99d0-4410-a2ff-f694a896a035]
+    external_ids        : {name=egressDefaultDeny}
+    name                : egressDefaultDeny
+    ports               : []
+
+```
+
+Two ACLs (four total) are added to each PortGroup:
+
+1. a drop policy with `priority=1000` and `direction=to-lport`
+
+```
+   _uuid               : f174fcf1-a7c2-496d-9b96-136aaccc014f
+    action              : drop
+    direction           : to-lport
+    external_ids        : {default-deny-policy-type=Ingress}
+    log                 : false
+    match               : "outport == @ingressDefaultDeny"
+    meter               : []
+    name                : []
+    priority            : 1000
+    severity            : []
+
+
+```
+
+2. an allow policy for ARP traffic with `priority=1001`, `direction=to-lport`, and `match=arp`
+
+```
+_uuid               : 3f864884-cdb7-44be-a60e-e4f743afc9d0
+action              : allow
+direction           : to-lport
+external_ids        : {default-deny-policy-type=Ingress}
+log                 : false
+match               : "outport == @ingressDefaultDeny && arp"
+meter               : []
+name                : []
+priority            : 1001
+severity            : []
+
+```
+
+## **Applying the network policy to specific pods using `spec.podSelector`**
+
+In some cases only certain pods in a namespace may need to be selected by the NetworkPolicy. To handle this feature the `spec.podSelector` field can be used as follows 
```

Comment:
nit: "by a NetworkPolicy" in my head at least the implies there can only be one in a namespace